### ISSUE TITLE
The ExportAttribute is now sealed

### ISSUE THREAD

**Issue by AndreSacilotto** [godotengine/godot#67624]
### Godot version

4.0 beta3

### System information

Windows

### Issue description

So in godot 3.X the export attribute was not sealed allowing to do neat things, drastically reducing the amount of code needed to be written.

As as example:
```csharp
[AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
public class ExportEnumFlagsAttribute : ExportAttribute
{
   public ExportEnumFlagsAttribute(Type enumType) : 
      base(PropertyHint.Flags, enumType.IsEnum ? string.Join(",", Enum.GetValues(enumType).Cast<int>()) : "Invalid Enum Type")
      { }
}
```
Would it be possible to remove the sealed modifier? Or it was put there because of now godot uses source generators and doing such things are no longer possible, in such approach.

### Steps to reproduce

Try to extend the ExportAttribute type

### Minimal reproduction project

-

**Comment by raulsntos** [comment]
> Or it was put there because of now godot uses source generators and doing such things are no longer possible, in such approach.

Yes that's what it is, since now we use source generators the attribute instance is no longer used, the generator just finds where the ExportAttribute constructor is used so even if it wasn't sealed, derived classes would have no effect.

**Comment by Flavelius** [comment]
I was also using this 'feature' extensively. If we can't do that ourselves now, then there should be some built-in replacement, like source-generated Export Attribute extensions, like ExportAsEnum, ExportAsEnumFlags or ExportAsRangedNumber, as there's otherwise quite some unnecessary typing involved, especially for the enum case

**Comment by raulsntos** [comment]
We'll likely try to match the same annotations that are available in GDScript.
